ts/components/DetailedlistItemComponent.tsx

```diff
@@ -106,14 +106,31 @@ const ICON_WIDTH = 24;
 /**
  * A component to display a touchable list item
  */
-export default class DetailedlistItemComponent extends React.PureComponent<Props> {
+export default class DetailedlistItemComponent extends React.Component<Props> {
   private getIconName = () =>
     this.props.isSelectionModeEnabled
       ? this.props.isItemSelected
         ? "io-checkbox-on"
         : "io-checkbox-off"
       : "io-right";
 
+  shouldComponentUpdate(nextProps: Readonly<Props>): boolean {
+    // assuming that:
+    //  - messages are immutable
+    //  - if the component is somehow reused the text content changes (avoid stale callbacks)
+    return (
+      this.props.isPaid !== nextProps.isPaid ||
+      this.props.isNew !== nextProps.isNew ||
+      this.props.isSelectionModeEnabled !== nextProps.isSelectionModeEnabled ||
+      this.props.isItemSelected !== nextProps.isItemSelected ||
+      this.props.isArchived !== nextProps.isArchived ||
+      this.props.text3 !== nextProps.text3 ||
+      this.props.text2 !== nextProps.text2 ||
+      this.props.text12 !== nextProps.text12 ||
+      this.props.text11 !== nextProps.text11
+    );
+  }
+
```

Comment:
This improves a lot the performance and now the wallet is again responsive, but since the `onPress` and `onLongPress` of the `MessageListItem` are created each time, it keeps rendering:

<img width="296" alt="Schermata 2021-12-06 alle 12 05 36" src="https://user-images.githubusercontent.com/26501317/144835542-69ad860b-e406-4bc0-b361-bf473ca5f682.png">


Since the `onPress`action is always the same (coming with drilling from `MessagesHomeScreen` but the navigation action could be also dispatched from here), as improvement we can move the rendering check from `DetailedlistItemComponent` to `MessageListItem`, using:
```typescript
const MessageListItem = ({}) => [...];

const MessageListItemMemo = React.memo(
  MessageListItem,
  (prev: Props, curr: Props) =>
    curr.message.id === prev.message.id &&
    curr.isRead === prev.isRead &&
    curr.isArchived === prev.isArchived &&
    curr.isSelectionModeEnabled === prev.isSelectionModeEnabled &&
    curr.isSelected === prev.isSelected &&
    curr.hasPaidBadge === prev.hasPaidBadge
);

export default MessageListItemMemo;
```
In this way we can avoid another rendering and improve the first wallet rendering, for example, from `136.1 ms` to `73.1 ms`. Do you think could be an improvement?

Reply:
Extracted to [MessageList/Item](https://github.com/pagopa/io-app/pull/3540/files#diff-1ffd7d368166536f89837d90dbaf4381bc4a0b75bc8b2dfa5a7c879e6c063cfaR189) as per our chat.
I left the same logic to `DetailedlistItemComponent` to keep the improvement in the wallet and removing only some unused styles.